aws-redshift-clusterparametergroup/src/main/java/software/amazon/redshift/clusterparametergroup/UpdateHandler.java

```diff
@@ -36,47 +44,110 @@ protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+                        .handleError(this::operateTagsErrorHandler)
+                        .done((tagsRequest, tagsResponse, client, model, context) -> ProgressEvent.<ResourceModel, CallbackContext>builder()
+                                .callbackContext(callbackContext)
+                                .callbackDelaySeconds(0)
+                                .resourceModel(request.getDesiredResourceState())
+                                .status(OperationStatus.IN_PROGRESS)
+                                .build()))
 
-                .then(progress -> proxy.initiate("AWS-Redshift-ClusterParameterGroup::Update::ResetInstance", proxyClient, progress.getResourceModel(), progress.getCallbackContext())
+                .then(progress -> proxy.initiate("AWS-RedshiftServerless-Workgroup::Update::ReadParameters", proxyClient, progress.getResourceModel(), progress.getCallbackContext())
+                        .translateToServiceRequest(Translator::translateToReadParametersRequest)
+                        .makeServiceCall(this::describeClusterParameters)
+                        .handleError(this::describeClusterParametersErrorHandler)
+                        .done((readRequest, readResponse, client, model, context) -> ProgressEvent.<ResourceModel, CallbackContext>builder()
+                                .callbackContext(context)
+                                .callbackDelaySeconds(0)
+                                .resourceModel(getUpdatableResourceModel(model, Translator.translateFromReadParametersResponse(readResponse, model)))
+                                .status(OperationStatus.IN_PROGRESS)
+                                .build()))
+
+                .then(progress -> proxy.initiate("AWS-Redshift-ClusterParameterGroup::Update::ResetParameters", proxyClient, progress.getResourceModel(), progress.getCallbackContext())
                         .translateToServiceRequest(Translator::translateToResetRequest)
                         .makeServiceCall(this::resetClusterParameterGroup)
                         .handleError(this::resetClusterParameterGroupErrorHandler)
                         .progress()
                 )
 
-                .then(progress -> CollectionUtils.isEmpty(progress.getResourceModel().getParameters()) ? progress :
-                        proxy.initiate("AWS-Redshift-ClusterParameterGroup::Update::UpdateInstance", proxyClient, progress.getResourceModel(), progress.getCallbackContext())
-                                .translateToServiceRequest(Translator::translateToUpdateRequest)
-                                .makeServiceCall(this::modifyClusterParameterGroup)
-                                .handleError(this::modifyClusterParameterGroupErrorHandler)
-                                .progress())
+                .then(progress -> proxy.initiate("AWS-Redshift-ClusterParameterGroup::Update::UpdateParameters", proxyClient, progress.getResourceModel(), progress.getCallbackContext())
```

Comment:
do we now use UpdateParameters api ?

Reply:
we use `modifyClusterParameterGroup` api